File: php/features.php
<?php
declare(strict_types=1);

// Amiiboapi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AmiiboapiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AmiiboapiBaseFeature();
            case "test":
                return new AmiiboapiTestFeature();
            default:
                return new AmiiboapiBaseFeature();
        }
    }
}

File: py/features.py
# Amiiboapi SDK feature factory

from feature.base_feature import AmiiboapiBaseFeature
from feature.test_feature import AmiiboapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AmiiboapiBaseFeature(),
        "test": lambda: AmiiboapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
